src/System.Memory/src/System/Memory.cs

```diff
@@ -246,7 +247,20 @@ public bool TryGetArray(out ArraySegment<T> arraySegment)
         /// allocates, so should generally be avoided, however it is sometimes
         /// necessary to bridge the gap with APIs written in terms of arrays.
         /// </summary>
-        public T[] ToArray() => Span.ToArray();
+        public T[] ToArray()
+        {
```

Comment:
What was wrong with `Span.ToArray()` ?

Reply:
Going through span causes issues with the debugger view.
![image](https://user-images.githubusercontent.com/6527137/30769422-b42fa256-9fcc-11e7-98fc-3b02d946c5aa.png)
On retry, I get:
> Error = Unable to evaluate the expression. Operation not supported. Unknown error: 0x80070057.


I tried to fix that by calling Retain() to get a pointer instead of ToArray() as follows, but it doesn't work. **Any ideas how to resolve this issue?**
> Error = Evaluation of method System.MemoryDebugView`1[System.Byte].get_Items() calls into native method System.Runtime.InteropServices.GCHandle.InternalAlloc(). Evaluation of native methods in this context is not supported.
```C#
            get
            {
                T[] result = new T[_memory.Length];

                MemoryHandle handle = _memory.Retain(pin: true);

                for (int i = 0; i < result.Length; i++)
                {
                    result[i] = Unsafe.Read<T>(Unsafe.Add<T>(handle.PinnedPointer, i));
                }

                handle.Dispose();
                return result;
            }
```
![image](https://user-images.githubusercontent.com/6527137/30769243-4208a24c-9fca-11e7-93e9-62148b397e31.png)

Reply:
> Going through span causes issues with the debugger view

Could you please get debugger folks to take a look before starting on these workarounds? They maybe just hiding the problem, without actually solving.

Does the debugger view work for Memory backed by unmanaged memory?

Reply:
> Could you please get debugger folks to take a look before starting on these workarounds? They maybe just hiding the problem, without actually solving.

Will do.

> Does the debugger view work for Memory backed by unmanaged memory?

No it doesn't work (since it goes back to using Span):
![image](https://user-images.githubusercontent.com/6527137/30770331-b3a156b6-9fe2-11e7-936b-18603ccb980f.png)

Since there is no way for me to construct a Memory out of a pointer yet (we don't yet provide a concrete OwnedMemory that works on pointers), I used the NativeOwnedMemory impl provided here: https://github.com/dotnet/corefx/blob/master/src/Common/tests/System/Buffers/NativeOwnedMemory.cs

Reply:
It's a fundamental problem with the debugger. I already talked to them. There is a bug filed. I can dig it out on Monday.

Reply:
This needs comments & links to the bug. Can the workaround be moved to the MemoryDebugView - does DangerousTryGetArray work?